doc/source/exclude_patterns.md

```diff
@@ -0,0 +1,44 @@
+# Exclude Pattern Settings 
+```--exclude``` is an optional command-line argument for [in-toto-run](https://in-toto.readthedocs.io/en/latest/command-line-tools/in-toto-run.html) that uses [PathSpec](http://python-path-specification.readthedocs.io) while ```exclude_pattern``` is an API argument that follows [gitignore-style patterns](https://git-scm.com/docs/gitignore) and may be specified via the ```ARTIFACT_EXCLUDE_PATTERNS``` setting.
+
+The feature prevents given paths and hashes of 'materials' (files before command execution) and 'products' (files after command execution) from being written together with other information (executed command, return value, stdout and stderr) in generating link metadata files. When a given pattern's directory is excluded, all its files and subdirectories are excluded as well.
+
+# Pattern Format
+  - A backslash must be included in front of patterns beginning with a literal hash or exclamation mark. 
+  - The slash is a directory seperator. Separators at beginning or middle (or both) of a pattern are relative to directory level of particular .gitignore file. Separators at the end of the pattern only  match directories. 
+  - Beginning double asterisks match any leading path segments, inner double asterisks match multiple (or zero) inner path segments, and ending double asterisks match any trailing path segments. Single asterisks match everything except a slash. 
+  - Question marks match any one character except a slash.
+
+## Special Patterns
+1. ### Pattern beginning with a slash ###
+  - Match paths on the root directly instead of any descendant paths by making a pattern relative to root. Escape the relative root conversion with an extra forward slash in front so as to match absolute paths. (i.e, "//"). 
+
+2. ### Pattern that is a slash ###
+  - According to 'git check-ignore' (v2.4.1), a single '/' does not match any file
```

Comment:
I wonder how we want to do this. My distro, for example, has a newer version of git. Can we point to the current gitignore documentation for these (while also mentioning them)?

Reply:
I've linked the current gitignore documentation in the documentation section, but it doesn't say anything about a single slash pattern. Furthermore, I can't seem to find the documentation for 'git check-ignore' (v2.4.1) to verify this but it's what the [Pathspec documentation](https://github.com/cpburnz/python-path-specification/blob/master/pathspec/patterns/gitwildmatch.py#L59-L61) describes.

Reply:
Yeah, I think we can remove references to `git check-ignore` which is a CLI tool to check against `.gitignore` for specific paths in the repository. Let's just state the fact and move on, and the links to the docs at the bottom should handle the rest, especially since we start the document naming both PathSpec and gitignore.